tests/extension/app/popup/popup.js

```diff
@@ -26,6 +26,19 @@ function start() {
   chrome.runtime.sendMessage({ type: 'getstate' }, ({ status }) => {
     setStatus(status);
   });
+  
+  // [e2e-testing]: Since puppeteer only monitors logs for popup.js, we need
+  // to listen for error messages from other js files
+  chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+    if (request.type === 'offscreen') {
+      console.log(request.message)
+    }
+  })
+  chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
```

Comment:
Is there any reason we don't have a single listener and a conditional `if (request.type === 'offscreen' || request.type === 'worker') {`?

Reply:
Good point, but removed this listener in favor of just throwing error in `offscreen.js`.